pkg/stub/handler.go

```diff
@@ -282,12 +377,17 @@ func (h *Handler) CreateDefaultResourceIfNeeded() error {
 			}
 			logrus.Println("got already exists error on create default, just going to forgo this time vs. retry")
 			// just return the raw error and initiate the sdk's requeue/ratelimiter setup
-			return err
 		}
-		return h.GoodConditionUpdate(srcfg, corev1.ConditionTrue, v1alpha1.SamplesExist)
+		h.deleteInProgress = false
+		return err
+	} else {
+		if deleteInProgress {
+			return h.processError(srcfg, v1alpha1.SamplesExist, corev1.ConditionUnknown, err, "issues waiting for delete to complete: %v")
```

Comment:
you can do this return immediately after the wait.Poll call if err comes back !nil.